recipes-azure-iot/azure-identityd/aziot-identityd.inc

```diff
@@ -63,6 +65,9 @@ do_compile() {
     sed -i 's/^RELEASE = 0$/RELEASE = 1/'       ${S}/Makefile
     sed -i -e 's/CARGO_TARGET = \(.*\)/CARGO_TARGET = ${TARGET_SYS}/g' ${S}/Makefile
 
+    # for rust toolchain >= 1.82: extend the linker library search path via RUSTFLAGS
+    sed -i -e 's/CARGO = \(.*\)/CARGO = VENDOR_PREFIX="$(VENDOR_PREFIX)" VENDOR_PKGCONFIG="$(VENDOR_PKGCONFIG)" RUSTFLAGS="$(RUSTFLAGS) -L $(CARGO_OUTPUT_ABSPATH)" \\/g' ${S}/Makefile
```

Comment:
you do replace the captured content of `CARGO` instead of reusing it, which may be a problem if the Makefile changes.

Reply:
we already patch the rust code. why did you decide against using `cargo:rustc-link-search`?

Reply:
cargo:rustc-link-search is more only a fix of the sub project. imho, my fix is more general..

Reply:
" my fix is more general.." 

we can leave it as is, but i have questions resp. comments:
- you don't want a build system to be more general, e.g. if there are multiple versions of a library to link you want to know explicitly which one you're linking. 
besides that, since we're on a recipe level, i don't see how it is more "general". can you explain that how it is more general?
- furthermore it's in a place where i don't expect it to be. e.g. normally we adapt cargo related adaptions in the 
include file of the cargo bitbake generated recipe.
- this particualar problem is solved by other projects via cargo:rustc-link-search.

Reply:
btw. regarding the refactored command? why more complicated then before? all you had to change was to use the captured string.

Reply:
handling changed to cargo:rustc-link-search mechanism in build.rs